curvlinops/inverse.py

```diff
@@ -197,3 +204,121 @@ def _matvec(self, x: ndarray) -> ndarray:
                 )
 
         return self._scale * result
+
+
+class KFACInverseLinearOperator(_InverseLinearOperator):
+    """Class to invert instances of the ``KFACLinearOperator``."""
+
+    def __init__(
+        self,
+        A: KFACLinearOperator,
+        damping: Optional[Tuple[float, float]] = None,
+        cache: bool = True,
+    ):
+        """Store the linear operator whose inverse should be represented.
+
+        Args:
+            A: ``KFACLinearOperator`` whose inverse is formed.
+            damping: Damping values for all input and gradient covariances.
+            cache: Whether to cache the inverses of the Kronecker factors.
+                Default: ``True``.
+
+        Raises:
+            ValueError: If the linear operator is not a ``KFACLinearOperator``.
+        """
+        if not isinstance(A, KFACLinearOperator):
+            raise ValueError(
+                "The input `A` must be an instance of `KFACLinearOperator`."
+            )
+        super().__init__(A.dtype, A.shape)
+        self._A = A
+        self._damping = damping
+        self._cache = cache
+        self._inverse_input_covariances: Dict[str, Tensor] = {}
+        self._inverse_gradient_covariances: Dict[str, Tensor] = {}
+
+    def _compute_or_get_cached_inverse(self, name: str) -> Tuple[Tensor, Tensor]:
+        """Invert the Kronecker factors of the KFACLinearOperator or retrieve them.
+
+        Args:
+            name: Name of the layer for which to invert Kronecker factors.
+
+        Returns:
+            Tuple of inverses of the input and gradient covariance Kronecker factors.
+        """
+        if name in self._inverse_input_covariances:
+            aaT_inv = self._inverse_input_covariances.get(name)
+            ggT_inv = self._inverse_gradient_covariances.get(name)
+        else:
+            aaT = self._A._input_covariances.get(name)
+            ggT = self._A._gradient_covariances.get(name)
+            damping_aaT = self._damping[0] if self._damping is not None else 0.0
+            aaT_inv = (
+                cholesky_inverse(
+                    cholesky(aaT + damping_aaT * eye(aaT.shape[0], device=aaT.device))
+                )
+                if aaT is not None
+                else None
+            )
+            damping_ggT = self._damping[1] if self._damping is not None else 0.0
+            ggT_inv = (
+                cholesky_inverse(
+                    cholesky(ggT + damping_ggT * eye(ggT.shape[0], device=ggT.device))
+                )
+                if ggT is not None
+                else None
+            )
+            if self._cache:
+                self._inverse_input_covariances[name] = aaT_inv
```

Comment:
Only if layer's weight is in params.

Reply:
See https://github.com/f-dangel/curvlinops/pull/69#discussion_r1480333199.